### ISSUE TITLE
Explicit regularization seems better than early stopping

### ISSUE THREAD

**Issue by matsen** [matsengrp/gcdyn#74]
Just a note that in my most recent commit 8ad8495c3b134d352711e947f45f3ba77002824c I added some Dropout in the MLP and stopped having a validation set.

I think the preliminary results look good,

![image](https://github.com/matsengrp/gcdyn/assets/112708/0ff1a25d-ec1b-4921-b6d0-faaf0729bc65)

Here's what it looks like with early stopping (30 epochs, which I selected by looking at validation loss in a run that had it on) and no regularization:

![image](https://github.com/matsengrp/gcdyn/assets/112708/c43f984f-9a8e-44b7-b54e-04e988890fec)

It seems to me that there could be a bias, especially on the upper end, from an undertrained network. So, moving forward, I suggest we do long training with some regularization. 

Note that I tried out some regularization like 

```python
layers.TimeDistributed(layers.Conv1D(filters=25, kernel_size=4, kernel_regularizer=keras.regularizers.l2(0.01))),
layers.TimeDistributed(layers.BatchNormalization()),
layers.TimeDistributed(layers.Activation(actfn)),
layers.TimeDistributed(layers.Conv1D(filters=25, kernel_size=4, kernel_regularizer=keras.regularizers.l2(0.01))),
layers.TimeDistributed(layers.BatchNormalization()),
layers.TimeDistributed(layers.Activation(actfn)),
layers.TimeDistributed(layers.MaxPooling1D(pool_size=2, strides=2)),  # Downsampling by a factor of 2
layers.TimeDistributed(layers.Conv1D(filters=40, kernel_size=4, kernel_regularizer=keras.regularizers.l2(0.01))),
layers.TimeDistributed(layers.BatchNormalization()),
layers.TimeDistributed(layers.Activation(actfn)),
layers.TimeDistributed(layers.GlobalAveragePooling1D()),
```

which didn't seem to help but we could keep it in mind.
